src/Symfony/Component/Form/Tests/Extension/Core/Type/IntegerTypeTest.php

```diff
@@ -24,6 +24,34 @@ protected function setUp(): void
         parent::setUp();
     }
 
+    public function testArabicLocale()
+    {
+        $default = \Locale::getDefault();
```

Comment:
This (and restoring the locale) should be done in `setUp()` and `tearDown()` respectively to make sure that this also happens when a test fails (currently in such a case the `\Locale::setDefault($default)` call would not be reached).

Reply:
But we need that only for this test case, and not all?

Reply:
To clarify: The `\Locale::setDefault('ar')` should still be part of the test case itself. So it should finally look like this:

```php
private $previousLocale;

protected function setUp(): void
{
    $this->previousLocale = \Locale::getDefault();

    // ..
}

protected function tearDown(): void
{
    \Locale::setDefault($this->previousLocale);

    // ...
}

public function testArabicLocale()
{
    \Locale::setDefault('ar');

    // ...
}
```